### PR TITLE
Fix #503: Migrate WebClient to rest-client-base

### LINKED ISSUE #503: Migrate WebClient to rest-client-base
Migrate the following classes to the `rest-client-base` library which provides a common implementation of the REST client:
- `CallbackUrlBehavior`
- `PowerAuthRestClient`

### PR DESCRIPTION
**Pull request by romanstrobl**

### CODE REVIEW

File: powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/behavior/tasks/v3/CallbackUrlBehavior.java [hunk 1]
```diff
@@ -91,9 +87,9 @@ public void setConfiguration(PowerAuthServiceConfiguration configuration) {
     }
 
     /**
-     * Configure WebClient HTTP connection parameters.
+     * Configure Rest client HTTP connection parameters.
      */
-    private void configureWebClient() {
+    private void configureRestClient() {
```
Comment (petrdvorak) [inline powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/behavior/tasks/v3/CallbackUrlBehavior.java:92]: Do we really need this method? It is only used once, right before initializing the client, and the only purpose is to store values from configuration to local properties, that I also think we no longer need.

**Review by petrdvorak — CHANGES_REQUESTED** [review]
There is some left-over code that probably could be cleaned up a bit - let's do it with this change...

File: powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/behavior/tasks/v3/CallbackUrlBehavior.java (hunk 1 above)
Comment (romanstrobl) [reply powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/behavior/tasks/v3/CallbackUrlBehavior.java:92]: True, we can set the configuration directly.

**Review by petrdvorak — APPROVED** [review]
Looks OK

**Pull request merged**